cloudsim-plus/src/main/java/org/cloudbus/cloudsim/brokers/DatacenterBrokerAbstract.java

```diff
@@ -771,14 +774,71 @@ public void requestIdleVmDestruction(final Vm vm) {
         }
     }
 
+    private Cloudlet extractCloudlet(CloudletExecution cloudletExecution) {
+        return cloudletExecution.getCloudlet();
+    }
+
+    private void resetCloudletStatus(final Cloudlet cloudlet) {
+        cloudlet.reset();
+    }
+
+    private Cloudlet extractResetCloudlet(CloudletExecution cloudletExecution) {
+        Cloudlet cloudlet = extractCloudlet(cloudletExecution);
+        this.resetCloudletStatus(cloudlet);
+        return cloudlet;
+    }
+
+    private List<Cloudlet> resetCloudlets(List<CloudletExecution> toReset) {
+        return toReset
+            .stream()
+            .map(this::extractResetCloudlet)
+            .collect(Collectors.toList());
+    }
+
+    private List<Cloudlet> resetCloudletsFromVm(final Vm vm) {
+        final List<Cloudlet> cloudletsToReschedule = new ArrayList<>();
```

Comment:
Probably the issue with your scenario with about ~60k cloudlets is here.
You are creating the ArrayList cloudletsToReschedule and adding elements to it.
When you call the first addAll, the list may be resized to accommodate all the new elements.
When you call addAll again, the list may be resized again.
Since we know exactly how many items will be added to the list, we can create it with the required capacity, avoiding the resize operation, as below:

```suggestion
        final int capacity = cloudletScheduler.getCloudletWaitingList().size() + cloudletScheduler.getCloudletExecList().size();
        final List<Cloudlet> cloudletsToReschedule = new ArrayList<>(capacity);
```